### PR TITLE
Check submodule with Github Actions

### PR DESCRIPTION
**Pull request by matinlotfali**

### CODE REVIEW

File: .github/workflows/check-submodule.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Check SHA hash of submodule and the master/develop branch
+        run: |
+          cd $GITHUB_WORKSPACE/Build/meta-mistysom
+          SOM_HEAD = $(git rev-parse HEAD)
+          echo "The submodule is set to $SOM_HEAD\n"
+          SOM_ORIGIN = $(git rev-parse origin/$GITHUB_REF_NAME)
+          echo "The origin/$GITHUB_REF_NAME is set to $SOM_ORIGIN"
+          [ $SOM_HEAD == SOM_ORIGIN ]
+        shell: bash
```
Comment (mistyron) [inline .github/workflows/check-submodule.yml:29]: will this fail if the branches misalign?

**Review by mistyron — COMMENTED** [review]
review comment

File: .github/workflows/check-submodule.yml (hunk 1 above)
Comment (matinlotfali) [reply .github/workflows/check-submodule.yml:29]: On paper yes. But it is failing before it reaches it. Still testing.

Comment (mistyron) [reply .github/workflows/check-submodule.yml:29]: Sounds good! Keep testing and let me know when it works! :)

**Review by mistyron — APPROVED** [review]
I approve with a note that this feature is still under testing - merging to `develop` would help with the  verification

**Review by mistyron — APPROVED** [review]
LGTM, does it work? :)

**Comment by matinlotfali** [comment]
Thanks to https://github.com/jtmullen/submodule-branch-check-action/issues/29 the issue is now resolved and the check is working.
@mistyron Please review the YAML file and its parameters.

**Review by mistyron — APPROVED** [review]
LGTM

**Pull request merged**